sources/api/netdog/src/networkd/config/mod.rs

```diff
@@ -7,7 +7,7 @@ use super::Result;
 pub(crate) use netdev::{NetDevBuilder, NetDevConfig};
 pub(crate) use network::{NetworkBuilder, NetworkConfig};
 
-const NETWORKD_CONFIG_DIR: &str = "/etc/systemd/network";
+const NETWORKD_CONFIG_DIR: &str = "/run/systemd/network";
```

Comment:
I don't love this change - there are a lot more protections on `/etc`. Why is this needed?

Reply:
Rather than writing to run, I've changed it back to `/etc/systemd/networkd`, and am creating that directory via mount unit with the correct context attached.

Reply:
I haven't done much yet with Bottlerocket's mounts, but isn't `/etc` already mounted as tmpfs? Why do we need an additional mount?

Reply:
Indeed `/etc` is mounted as a tmpfs.  We need the directory `/etc/systemd/network` to be created, but because `systemd-networkd` runs with `DefaultDependencies=no` (which includes local-fs), we can't use tmpfiles.d to help us there.  A mount unit also allows us to create the directory with the proper SELinux label.

I'm open to suggestions if there's a better way that handles the timing issue we have!